lib/ansible/module_utils/k8s_common.py

```diff
@@ -215,38 +782,57 @@ def execute_module(self):
             try:
                 self.helper.object_from_params(self.params, obj=k8s_obj)
             except KubernetesException as exc:
-                self.fail_json(msg="Failed to patch object: {}".format(exc.message))
+                self.fail_json(msg="Failed to patch object: {0}".format(exc.message))
             match, diff = self.helper.objects_match(existing, k8s_obj)
             if match:
-                return_attributes[self.kind] = existing.to_dict()
+                return_attributes['result'] = existing.to_dict()
                 self.exit_json(**return_attributes)
-            else:
-                self.log('Existing:')
-                self.log(json.dumps(existing.to_str(), indent=4))
-                self.log('\nDifferences:')
-                self.log(json.dumps(diff, indent=4))
+            elif self._diff:
+                return_attributes['differences'] = diff
             # Differences exist between the existing obj and requested params
             if not self.check_mode:
                 try:
                     k8s_obj = self.helper.patch_object(name, namespace, k8s_obj)
                 except KubernetesException as exc:
-                    self.fail_json(msg="Failed to patch object: {}".format(exc.message))
-            return_attributes[self.kind] = k8s_obj.to_dict()
+                    self.fail_json(msg="Failed to patch object: {0}".format(exc.message))
+            return_attributes['result'] = k8s_obj.to_dict()
             return_attributes['changed'] = True
             self.exit_json(**return_attributes)
 
+    def _authenticate(self):
```

Comment:
It would be great if we could support authentication from the POD as well. I've some scenarios where I'm running `k8s` tasks from a `k8s` POD and I need to access the `k8s` API - to create a configmap, for example.

I've a script that does this:

```python
     try:
         config.load_incluster_config()
     except config.config_exception.ConfigException:
         config.load_kube_config()
```

Wonder if we could do something similar. Try to load the `incluster` config and then fallback to loading the kube config file. All this, of course, if no specific configs have been specified in the task itself.

Reply:
@flaper87 

This is new to me. I didn't know about `load_incluster_config`. Thank you for  pointing it out! 

Should this be driven by the user? In other words, should there be an option like `load_incluster_config`, that if True, will cause the module to ignore the usual config path, and instead call `load_incluster_config`?

I'm not opposed to your example above, just wondering what's the best path. I don't have experience with a use case for this option. I can see the benefit of the try/except approach, where the module just handles things. But, on the other hand, it seems safer to have the user be explicit.